src/frontend/src/common/redux/index.ts

```diff
@@ -1,12 +1,33 @@
 import { AnyAction, applyMiddleware, createStore } from "redux";
 import { composeWithDevTools } from "redux-devtools-extension";
+import { fetchUserInfo } from "../queries/auth";
 import { getLocalStorage } from "../utils/localStorage";
-import { setGroupMiddleware } from "./middleware";
+import { setGroup, setPathogen } from "./actions";
+import { setGroupMiddleware, setPathogenMiddleware } from "./middleware";
 import { CZGEReduxActions, Pathogen, ReduxPersistenceTokens } from "./types";
 
+/**
+ * A note about how our redux store is initialized ...
+ * Before we create the store, we make an initial state object.
+ * We try to read from localstorage to create the initial state.
+ * If we can't, we use reasonable defaults that match the types we want
+ * each state field to have.
+ * We create the store with the initial state, also registering middleware with it.
+ * Then we set "true" defaults.
+ * Why set defaults after instead of in initial state?
+ * Because otherwise those updates cannot be persisted to localstorage using the predefined
+ * middleware, since the store doesn't actually exist yet.
+ * Although it would be possible to make requests to calculate and manually store the data,
+ * this request would be async, meaning the state would not reliably be set before the ui
+ * comes online.
+ * So we have a compromise to use defaults for a fast initial hydration, and then immediately
+ * make reqeusts for "true" default values from the server if needed.
+ */
+
+// first, load state from localstorage if any exists and use it to initialize redux
 const getInitialState = () => {
   const storedGroupStr = getLocalStorage(ReduxPersistenceTokens.GROUP);
-  const storedGroup = storedGroupStr ? parseInt(storedGroupStr) : null;
+  const storedGroup = storedGroupStr ? parseInt(storedGroupStr) : -1;
```

Comment:
Nit: I generally prefer `null` as a way of informing that something does not exist at all rather than using `-1` or `""`. I know for interfaces it means that there's a bit of extra boilerplate since you need to union on a `| null` possibility, but it feels -- to me at least -- more obvious what the interface is representing. With just a `key: number` I assume the property is always present and known, while with `key: number | null` it feels more clear that it can also be in a "weird" state (`key?: number` also accomplishing the same, albeit with different undertones).

Reply:
I don't want it to be possible to ever set the group to `null`, though. I prefer to load it in as a placeholder value for a second while the app loads and always have it be a number from that point on.